rdkPlugins/Logging/source/NullSink.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2022 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "NullSink.h"
+
+#include <unistd.h>
+#include <strings.h>
+#include <fcntl.h>
+#include <map>
+#include <sys/stat.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+#include <Logging.h>
+
+NullSink::NullSink(const std::string &containerId, std::shared_ptr<rt_dobby_schema> &containerConfig)
+    : mContainerConfig(containerConfig),
+      mContainerId(containerId),
+      mBuf{}
+{
+    AI_LOG_FN_ENTRY();
+
+    mDevNullFd = open("/dev/null", O_CLOEXEC | O_WRONLY);
+    if (mDevNullFd < 0)
+    {
+        AI_LOG_SYS_ERROR(errno, "Failed to open /dev/null");
+    }
+
+    AI_LOG_FN_EXIT();
+}
+
+NullSink::~NullSink()
+{
+    if (mDevNullFd > 0)
+    {
+        if (close(mDevNullFd) < 0)
+        {
+            AI_LOG_SYS_ERROR(errno, "Failed to close journald stream");
+        }
+    }
+}
+
+void NullSink::SetLogOptions(const IDobbyRdkLoggingPlugin::LoggingOptions &options)
+{
+    std::lock_guard<std::mutex> locker(mLock);
+    mLoggingOptions = options;
+}
+
+void NullSink::DumpLog(const int bufferFd)
+{
+    std::lock_guard<std::mutex> locker(mLock);
+
+    ssize_t ret;
+    memset(mBuf, 0, sizeof(mBuf));
```

Comment:
Small optimization - memset doesn't need to be part of critical section. You can move mutex below this line, so buffer can be acquired even while other thread is holding the mutex.